lib/chef/resource/chef_client_scheduled_task.rb

```diff
@@ -149,9 +172,9 @@ class ChefClientScheduledTask < Chef::Resource
           password                       new_resource.password
           frequency                      new_resource.frequency.to_sym
           frequency_modifier             new_resource.frequency_modifier if frequency_supports_frequency_modifier?
-          start_time                     new_resource.start_time
+          start_time                     start_time_value
```

Comment:
I'm not clear why its necessary to have this since you can manually set the start_time to the top of any hour in the past.  The scribbling over the value on every subsequent run makes me a bit nervous as well, it'd be better to have a system where the windows_task resource could be idempotent on every run.

Reply:
The reason to have the snapping feature is to have the task on a predictable cadence.  However the idempotency issue is something I overlooked.  It would make more sense for the snap_time to be a boolean property of the windows_task itself and as long as the start time conforms to the snapping algorithm it would not need to be updated, solving the idempotency issue.  I will submit that there, and leave the consistent splay here.